artifact/tar/platform/libarchive/tar.cpp

```diff
@@ -0,0 +1,129 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include <cstdint>
+#include <vector>
+#include <algorithm>
+#include <string>
+#include <iostream>
+
+#include <common/io.hpp>
+#include <common/log.hpp>
+#include <artifact/tar/tar.hpp>
+
+namespace mender {
+namespace tar {
+
+using namespace std;
+
+namespace log = mender::common::log;
+
+const ErrorCategoryClass ErrorCategory {};
+
+const char *ErrorCategoryClass::name() const noexcept {
+	return "TarErrorCategory";
+}
+
+string ErrorCategoryClass::message(int code) const {
+	switch (code) {
+	case NoError:
+		return "Success";
+	case TarReaderError:
+		return "Error reading the tar stream";
+	case TarShortReadError:
+		return "Short read error";
+	case TarEntryError:
+		return "Error reading the tar entry";
+	case TarEOFError:
+		return "Archive EOF reached";
+	default:
+		return "Unknown";
```

Comment:
I know this isn't consistent in the code base, but I suggest switching to [this style](https://github.com/mendersoftware/mender/blob/feature-c%2B%2B-client/common/http/http.cpp#L42) instead, which should help us catch missing cases.

Reply:
Still not quite what I was looking for. You need to move the assert and return outside the `switch` block and remove the `default`. The intention is to have the compiler warn us when there is a enum value which is isn't covered, but the `default` catch-all prevents that.

Reply:
Ahh, alright. With you! Fixed